### PR TITLE
Add Automation subsection

### PR DESCRIPTION
**Pull request by bwalderman**
This change updates the spec with a WebDriver extension command for modifying the user agent's storage access policy. This is to address: https://github.com/w3c/webdriver/issues/1437#issue-474647752

### CODE REVIEW

File: storage-access.bs [hunk 1]
```diff
@@ -293,6 +307,42 @@ ISSUE: Write this section.
...
+The <dfn export>Set Storage Access</dfn> [=extension command=] modifies the storage access policy for the [=current browsing context=].
+
+The [=remote end steps=] are:
+
+1. If |parameters| is not a JSON [=Object=], return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+1. Let |blocked| be the result of [=getting a property=] from |parameters| named `blocked`.
+1. If |blocked| is not a [=boolean=] return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+1. Let |doc| be the [=current browsing context=]'s [=active document=].
+1. Let |key| be the result of [=generate a partitioned storage key|generating a partitioned storage key=] from |doc|.
+1. If |key| is failure, return a [=WebDriver error=] with [=WebDriver error code=] [=unknown error=].
```
Comment (burg) [inline storage-access.bs:338]: Nit: I prefer "unsupported operation" over "unknown error". This is only likely to fail if the remote end (say, WebKit) doesn't support the operation.

Comment (bwalderman) [reply storage-access.bs]: Thanks. I was on the fence about this myself. Switching to "unsupported operation".

File: storage-access.bs [hunk 2]
```diff
@@ -293,6 +307,42 @@ ISSUE: Write this section.
...
+        <td>/session/{session id}/storageaccess</td>
+    </tr>
+    </tbody>
+</table>
+
+The <dfn export>Set Storage Access</dfn> [=extension command=] modifies the storage access policy for the [=current browsing context=].
+
+The [=remote end steps=] are:
+
+1. If |parameters| is not a JSON [=Object=], return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
```
[the remark this answers is not in the document]
Comment (bwalderman) [reply storage-access.bs]: This is already covered by https://w3c.github.io/webdriver/#processing-model (see step 5 where  `parameters` is parsed) so I've removed this check.

**Comment by bwalderman** [comment]
> What should happen if the browsing context is the top-level browsing context? What if its active document is same-site with the top-level site?

Thanks, I've added steps to handle these cases.

> It seems there should also be API surface to ensure certain outcomes before documents are created. That is, that you know what when you create a nested document it won't have partitioned data (or will).

I've changed the API slightly. It now treats the current browsing context as the top-level site, and take a parameter to specify which embedded site to block or allow. This makes it possible to set policy for an embedded site without needing to create any nested documents and switch contexts first.

File: storage-access.bs [hunk 3]
```diff
@@ -293,6 +308,49 @@ ISSUE: Write this section.
...
+
+The [=remote end steps=] are:
+
+
+1. Let |blocked| be the result of [=getting a property=] from |parameters| named `blocked`.
+1. If |blocked| is not a [=boolean=] return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+1. Let |origin| be the result of [=getting a property=] from |parameters| named `origin`.
+1. If |origin| is not the [=serialization of an origin=] return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+1. If |origin| is an [=opaque origin=] return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+1. Let |embedded site| be the result of [=obtain a site|obtaining a site=] from |origin|.
```
Comment (Brandr0id) [inline storage-access.bs:340]: Adding the ability to specify this per origin seems useful. Would it be useful to allow a mechanism to specify all origins (e.g. equiv to "blockallthirdpartycookies" preferences) as well?

File: storage-access.bs (hunk 2 above)
[the remark this answers is not in the document]
Comment (annevk) [reply storage-access.bs]: Thanks, I filed https://github.com/w3c/webdriver/issues/1529 to see about improving that further.

File: storage-access.bs [hunk 4]
```diff
@@ -293,6 +308,49 @@ ISSUE: Write this section.
...
+
+The <dfn export>Set Storage Access</dfn> [=extension command=] modifies the storage access policy for the [=current browsing context=].
+
+The [=remote end steps=] are:
+
+
+1. Let |blocked| be the result of [=getting a property=] from |parameters| named `blocked`.
+1. If |blocked| is not a [=boolean=] return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+1. Let |origin| be the result of [=getting a property=] from |parameters| named `origin`.
+1. If |origin| is not the [=serialization of an origin=] return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
```
Comment (annevk) [inline storage-access.bs:338]: This doesn't work. I guess what you want to be doing is parse this field as a URL and then use that URL's origin. Similar to `postMessage()`. That would also help the following steps which assume origin is an object (and not a string).

Comment (bwalderman) [reply storage-access.bs]: Thanks. I've adapted some of the language from the spec for `postMessage`. I've also added similar handling for asterisk (*) to provide a mechanism to specify all origins per @Brandr0id's request above.

**Pull request merged**